sys/include/net/gnrc/tcp/tcb.h

```diff
@@ -47,14 +47,12 @@ extern "C" {
  * @brief Transmission control block of GNRC TCP.
  */
 typedef struct _transmission_control_block {
-    uint8_t address_family;                   /**< Address Family of local_addr / peer_addr */
-#ifdef MODULE_GNRC_IPV6
-    uint8_t local_addr[sizeof(ipv6_addr_t)];  /**< Local IP address */
-    uint8_t peer_addr[sizeof(ipv6_addr_t)];   /**< Peer IP address */
-    int8_t  ll_iface;                         /**< Link layer interface id to use. */
-#endif
-    uint16_t local_port;   /**< Local connections port number */
-    uint16_t peer_port;    /**< Peer connections port number */
+    int *address_family;       /**< Address Family in use */
+    uint8_t *local_addr;       /**< Pointer to local address storage */
+    uint16_t *local_port;      /**< Pointer to local port storage */
+    uint8_t *peer_addr;        /**< Pointer to peer address storage */
+    uint16_t *peer_port;       /**< Pointer to peer port storage */
+    int8_t *ll_iface;          /**< Pointer to link-layer interface storage */
```

Comment:
There are some members here that clearly should be scalars.

Reply:
Well the only thing I did here was to externalize all data, that is stored in sock_ep_t. 
Things like address_familiy and ll_iface could be stored locally, however I would like to use the Pointers to keep the information stored in sock_ep_t in sync with interally used values.

Reply:
Shouldn't this API rather stand on its own feet? If you want to make it transparent to `sock_ep_t` use a type transparent to `sock_ep_t`. Don't force a user to use `sock` just because your API is not usable without it.

Reply:
I though the hole Idea of sock is to unify access to different stacks under on interface. I don't see much benefits of using gnrc_tcp without sock (except a slightly smaller memory footprint).

But as mentioned above: Providing a dedicated struct would lead to a cleaner interface.

Reply:
> I don't see much benefits of using gnrc_tcp without sock (except a slightly smaller memory footprint).

That's exactly the benefit I was thinking about ;-)